File: 13 - Determining Compund Interest/Golang/main.go
package main

import (
	"fmt"
	"math"
)

func getNum(s string) float64 {
	var t float64
	fmt.Printf("What is the %s? ", s)
	fmt.Scanf("%f", &t)
	return t
}

func main() {
	P := getNum("principal amount")
	r := getNum("annual rate of interest")
	t := getNum("number of years the amount is invested")
	n := getNum("number of times the investment is compounded per year")

	fmt.Printf("£%.2f invested at %.2f%% for %.0f years compounded %.1f times per year is £%.2f\n",
		P, r, t, n, P*math.Pow((1+(r/100/n)), n*t))
}

File: 13 - Determining Compund Interest/C++/main.cpp
#include <iostream>
#include <cmath>

using namespace std;

float inputf(string output)
{
    cout << output;
    float t;
    cin >> t;
    return t;
}

int main(void)
{
    float principal, rate, time, compounded_per_year = 0;

    principal           = inputf("Enter the principal: £");
    rate                = inputf("Enter the rate: %");
    time                = inputf("Enter the number of years: ");
    compounded_per_year = inputf("Enter the number of times the investment is compounded per year: ");

    cout << "The investment will be worth: £"
         << principal * powf( (1 + (rate/100)/compounded_per_year), compounded_per_year * time )
         << endl;

}
